File: array/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {

        // init
        int result = 0;

        // use XOR to cancel out duplicate numbers
        for (int i = 0; i < nums.size(); i++) {
            result ^= nums[i];
        }

        // return the single number
        return result;
    }
};

File: bit-manipulation/single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # init
        result = 0

        # use XOR to cancel out duplicate numbers
        for num in nums:
            result ^= num

        # return the single number
        return result